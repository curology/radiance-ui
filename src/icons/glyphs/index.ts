// @create-index

export { default as AcneBodySprayGlyph } from './acneBodySprayGlyph';
export { default as AcneBodyWashGlyph } from './acneBodyWashGlyph';
export { default as AcneFiveGlyph } from './acneFiveGlyph';
export { default as AcneGlyph } from './acneGlyph';
export { default as AcneOneGlyph } from './acneOneGlyph';
export { default as AcneSevenGlyph } from './acneSevenGlyph';
export { default as AcneThreeGlyph } from './acneThreeGlyph';
export { default as AhaBhaTonerSolutionGlyph } from './ahaBhaTonerSolutionGlyph';
export { default as AntiAgingAndAcneGlyph } from './antiAgingAndAcneGlyph';
export { default as AntiAgingGlyph } from './antiAgingGlyph';
export { default as AzelaicAcidGlyph } from './azelaicAcidGlyph';
export { default as BodySunscreenGlyph } from './bodySunscreenGlyph';
export { default as BreakGlyph } from './breakGlyph';
export { default as CheckBadgeGlyph } from './checkBadgeGlyph';
export { default as ChestGlyph } from './chestGlyph';
export { default as CleanserGlyph } from './cleanserGlyph';
export { default as ClindamycinGlyph } from './clindamycinGlyph';
export { default as ClockAlwaysGlyph } from './clockAlwaysGlyph';
export { default as ClockMostOfTheTimeGlyph } from './clockMostOfTheTimeGlyph';
export { default as ClockNeverGlyph } from './clockNeverGlyph';
export { default as ClockSometimesGlyph } from './clockSometimesGlyph';
export { default as CloggedPoresGlyph } from './cloggedPoresGlyph';
export { default as CombinationGlyph } from './combinationGlyph';
export { default as ConsultHistoryGlyph } from './consultHistoryGlyph';
export { default as CourseGlyph } from './courseGlyph';
export { default as CycleGlyph } from './cycleGlyph';
export { default as DarkSpotFormulaGlyph } from './darkSpotFormulaGlyph';
export { default as DarkSpotsGlyph } from './darkSpotsGlyph';
export { default as DesonideGlyph } from './desonideGlyph';
export { default as DexpanthenolGlyph } from './dexpanthenolGlyph';
export { default as DropperGlyph } from './dropperGlyph';
export { default as ExerciseGlyph } from './exerciseGlyph';
export { default as ExfoliatingMaskGlyph } from './exfoliatingMaskGlyph';
export { default as EyecreamGlyph } from './eyecreamGlyph';
export { default as FaceGlyph } from './faceGlyph';
export { default as FemaleGlyph } from './femaleGlyph';
export { default as FineLinesGlyph } from './fineLinesGlyph';
export { default as FirmnessGlyph } from './firmnessGlyph';
export { default as FlagCanadaGlyph } from './flagCanadaGlyph';
export { default as FlagUkGlyph } from './flagUKGlyph';
export { default as FlagUsaGlyph } from './flagUSAGlyph';
export { default as FormulaChangeGlyph } from './formulaChangeGlyph';
export { default as GlobeGlyph } from './globeGlyph';
export { default as GreenTeaGlyph } from './greenTeaGlyph';
export { default as HandWithCreamGlyph } from './handWithCreamGlyph';
export { default as HydratingMaskGlyph } from './hydratingMaskGlyph';
export { default as HydrocolloidBandageGlyph } from './hydrocolloidBandageGlyph';
export { default as HydrocortisoneGlyph } from './hydrocortisoneGlyph';
export { default as HydroquinoneCycleGlyph } from './hydroquinoneCycleGlyph';
export { default as HydroquinoneGlyph } from './hydroquinoneGlyph';
export { default as IntersexGlyph } from './intersexGlyph';
export { default as KojicAcidGlyph } from './kojicAcidGlyph';
export { default as LightbulbGlyph } from './lightbulbGlyph';
export { default as LipBalmGlyph } from './lipBalmGlyph';
export { default as LockGlyph } from './lockGlyph';
export { default as LunchGlyph } from './lunchGlyph';
export { default as MaleGlyph } from './maleGlyph';
export { default as MelasmaGlyph } from './melasmaGlyph';
export { default as MessageGlyph } from './messageGlyph';
export { default as MetronidazoleGlyph } from './metronidazoleGlyph';
export { default as MicellarWaterGlyph } from './micellarWaterGlyph';
export { default as MixIngredientsGlyph } from './mixIngredientsGlyph';
export { default as MoisturizerGlyph } from './moisturizerGlyph';
export { default as MoonGlyph } from './moonGlyph';
export { default as MortorGlyph } from './mortorGlyph';
export { default as NeckGlyph } from './neckGlyph';
export { default as NiacinamideGlyph } from './niacinamideGlyph';
export { default as NoMakeupGlyph } from './noMakeupGlyph';
export { default as NoneGlyph } from './noneGlyph';
export { default as OftenDryGlyph } from './oftenDryGlyph';
export { default as OftenOilyGlyph } from './oftenOilyGlyph';
export { default as OilCleanserGlyph } from './oilCleanserGlyph';
export { default as OtherGlyph } from './otherGlyph';
export { default as PackagingGlyph } from './packagingGlyph';
export { default as PalmTreeGlyph } from './palmTreeGlyph';
export { default as PillsGlyph } from './pillsGlyph';
export { default as RelaxGlyph } from './relaxGlyph';
export { default as ResveratrolGlyph } from './resveratrolGlyph';
export { default as RichMoisturizerGlyph } from './richMoisturizerGlyph';
export { default as RocketshipGlyph } from './rocketshipGlyph';
export { default as SelfieGlyph } from './selfieGlyph';
export { default as SleepGlyph } from './sleepGlyph';
export { default as StressGlyph } from './stressGlyph';
export { default as SunGlyph } from './sunGlyph';
export { default as SunMoonGlyph } from './sunMoonGlyph';
export { default as SunblockGlyph } from './sunblockGlyph';
export { default as SunscreenGlyph } from './sunscreenGlyph';
export { default as SuperbottleGlyph } from './superbottleGlyph';
export { default as SupportCenterGlyph } from './supportCenterGlyph';
export { default as TextureGlyph } from './textureGlyph';
export { default as TimeAlwaysGlyph } from './timeAlwaysGlyph';
export { default as TimeMostOfTheTimeGlyph } from './timeMostOfTheTimeGlyph';
export { default as TimeNeverGlyph } from './timeNeverGlyph';
export { default as TimeSometimesGlyph } from './timeSometimesGlyph';
export { default as TranexamicAcidGlyph } from './tranexamicAcidGlyph';
export { default as TretinoinGlyph } from './tretinoinGlyph';
export { default as VeryDryGlyph } from './veryDryGlyph';
export { default as VeryOilyGlyph } from './veryOilyGlyph';
export { default as VitaminCGlyph } from './vitaminCGlyph';
export { default as VitaminCSerumGlyph } from './vitaminCSerumGlyph';
export { default as WashFaceGlyph } from './washFaceGlyph';
export { default as WrinklesGlyph } from './wrinklesGlyph';
export { default as YesGlyph } from './yesGlyph';
export { default as ZincGlyph } from './zincGlyph';
export { default as ZincSoapBarGlyph } from './zincSoapBarGlyph';
