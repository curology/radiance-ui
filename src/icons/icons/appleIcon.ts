import { AppleIcon } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(AppleIcon, AppleIcon, props);
