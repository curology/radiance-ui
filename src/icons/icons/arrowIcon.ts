import { ArrowIcon } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(ArrowIcon, ArrowIcon, props);
