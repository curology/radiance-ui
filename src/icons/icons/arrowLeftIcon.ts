import { ArrowLeftIcon, ArrowLeftIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) =>
  useIcon(ArrowLeftIcon, ArrowLeftIconSecondary, props);
