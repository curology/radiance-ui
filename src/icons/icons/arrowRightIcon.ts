import { ArrowRightIcon, ArrowRightIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) =>
  useIcon(ArrowRightIcon, ArrowRightIconSecondary, props);
