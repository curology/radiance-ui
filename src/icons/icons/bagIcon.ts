import { BagIcon, BagIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(BagIcon, BagIconSecondary, props);
