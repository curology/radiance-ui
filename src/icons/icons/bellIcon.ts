import { BellIcon, BellIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) =>
  useIcon(BellIcon, BellIconSecondary, props);
