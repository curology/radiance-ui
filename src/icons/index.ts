/*
 * Note: this file has to be manually updated if we add more types of icons.
 * We do not generate it via create-index
 */

export * from './emojis';
export * from './glyphs';
export * from './icons';
export * from './logos';
export * from './navIcons';
