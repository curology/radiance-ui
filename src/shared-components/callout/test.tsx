import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { NeckGlyph } from '../../icons';

import { Callout } from './index';

describe('<Callout />', () => {
  describe('UI snapshots', () => {
    it('renders a simple callout', () => {
      const { container } = render(
        <Callout>
          Simple callout text. <strong>Bold text supported</strong>
        </Callout>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
    it('renders callout with icon', () => {
      const { container } = render(
        <Callout icon={<NeckGlyph />}>
          Callout text with <strong>custom color</strong> and icon.
        </Callout>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });
});
