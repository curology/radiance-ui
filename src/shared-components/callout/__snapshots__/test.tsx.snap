// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Callout /> UI snapshots renders a simple callout 1`] = `
.emotion-0 {
  background-color: #EEEDF7;
  padding: 1rem;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  color: #332E54;
  font-size: 0.875rem;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    Simple callout text. 
    <strong>
      Bold text supported
    </strong>
  </div>
</div>
`;

exports[`<Callout /> UI snapshots renders callout with icon 1`] = `
.emotion-0 {
  background-color: #EEEDF7;
  padding: 1rem;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  color: #332E54;
  font-size: 0.875rem;
}

.emotion-4 {
  margin-left: 0.5rem;
}

.emotion-4 svg {
  fill: #332E54;
}

.emotion-4 svg path {
  fill: #332E54;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    Callout text with 
    <strong>
      custom color
    </strong>
     and icon.
  </div>
  <div
    class="emotion-4 emotion-5"
  >
    <svg
      class="emotion-6"
      fill="none"
      height="48"
      viewBox="0 0 48 48"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M36.983 20.387v-.267A3.746 3.746 0 0040 16.453c0-1.821-1.3-3.33-3.017-3.666V11h-1.45v9.387c0 3.376-6.869 7.6-9.56 8.968a3.829 3.829 0 01-3.992-.023c-2.645-1.346-9.514-5.57-9.514-8.945V11h-1.45v1.787A3.746 3.746 0 008 16.453c0 1.81 1.3 3.33 3.017 3.667v.267c0 2.657 2.738 5.279 5.407 7.216.15 1.358.208 4.096-1.404 6.15-1.15 1.462-2.96 2.309-5.372 2.54l.139 1.44c2.83-.267 4.977-1.3 6.381-3.087 1.532-1.938 1.81-4.316 1.787-6.01 1.624 1.056 2.994 1.764 3.342 1.95a5.259 5.259 0 005.406.034c.337-.174 1.52-.789 2.97-1.706-.034 1.694.28 3.899 1.718 5.732 1.404 1.787 3.585 2.82 6.486 3.086l.139-1.438c-2.483-.232-4.328-1.091-5.488-2.541-1.497-1.903-1.508-4.363-1.334-5.848 2.773-1.972 5.79-4.734 5.79-7.518zm1.567-3.934a2.28 2.28 0 01-1.567 2.158v-4.328a2.29 2.29 0 011.567 2.17zm-29.1 0c0-1.01.662-1.856 1.567-2.17v4.328a2.28 2.28 0 01-1.567-2.158z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
